app/Http/Controllers/TaskCategoriesController.php

```diff
@@ -41,14 +41,15 @@ public function create()
      * @return \Illuminate\Http\Response
      */
     public function store(TaskCategorySaveRequest $request)
-    {
+    {   
         $taskCategory = new TaskCategory;
         $taskCategory->user_id = $request->user()->id;
         $taskCategory->name = $request->name;
         $taskCategory->save();
 
-        return redirect()->route('taskCategory.index')
-            ->with('success', 'Data Added successfully!');
+        return response()->json('Category successfully added');
```

Comment:
The new category that was added should be returned in the response. Also, use [API Resources](https://laravel.com/docs/8.x/eloquent-resources#introduction) to construct the json schema of the response.